interop/nkl/parser.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Released under Apache 2.0 license as described in the file LICENSE.
+# Authors: Paul Govereau
+
+import types
+import inspect
+import ast
+import json
+
+from textwrap import dedent
+from itertools import chain
+from collections import deque
+
+class Enc(json.JSONEncoder):
+  def default(self, obj):
+    if isinstance(obj, ast.AST):
+      if len(obj.__dict__) == 0:
```

Comment:
This just seems like magic. Should we add comments here saying what is going on, or is it obvious? Comment applies to the whole file. Maybe an example as a comment would help the reader.

Reply:
I will add some comments. It is fairly opaque.